x-pack/dockerlogbeat/magefile.go

```diff
@@ -47,58 +57,155 @@ func getPluginName() (string, error) {
 	if err != nil {
 		return "", errors.Wrap(err, "error getting beats version")
 	}
-
 	return dockerPluginName + ":" + version, nil
 }
 
-// Build builds docker rootfs container root
-func Build() error {
-	mage.CreateDir(packageStagingDir)
-	mage.CreateDir(packageEndDir)
+// createContainer builds the plugin and creates the container that will later become the rootfs used by the plugin
+func createContainer(cli *client.Client) error {
+	gv, err := mage.GoVersion()
+	if err != nil {
+		return errors.Wrap(err, "error determining go version")
+	}
 
 	dockerLogBeatDir, err := os.Getwd()
 	if err != nil {
 		return errors.Wrap(err, "error getting work dir")
 	}
 
+	// change back to the root beats dir so we can send the proper build context to docker
 	err = os.Chdir("../..")
 	if err != nil {
 		return errors.Wrap(err, "error changing directory")
 	}
 
-	gv, err := mage.GoVersion()
+	// start to build the root container that'll be used to build the plugin
+	tmpDir, err := ioutil.TempDir("", "dockerBuildTar")
 	if err != nil {
-		return errors.Wrap(err, "error determining go version")
+		return errors.Wrap(err, "Error locating temp dir")
+	}
+	tarPath := filepath.Join(tmpDir, "tarRoot.tar")
+	err = sh.RunV("tar", "cf", tarPath, "./")
```

Comment:
We switch to the Beats root directory for this `tar` command?

Btw. stdlib also has the archive/tar package.

Reply:
Yah, we send the entire beats directory to the docker context to actually build the plugin, since we depend on libbeat and so on. Eventually I _really_ want to find a faster way to do this. Maybe just copy over `libbeat` to vendor during the build or something?